app/workers/activitypub/distribution_worker.rb

```diff
@@ -24,22 +26,42 @@ def perform(status_id)
   private
 
   def skip_distribution?
-    @status.direct_visibility? || @status.limited_visibility?
+    @status.direct_visibility?
+  end
+
+  def delegate_distribution?
+    @status.limited_visibility? && @status.thread&.limited_visibility? && !@status.thread&.account&.local?
   end
```

Comment:
This will break if two people on the same instance reply one after the other to a limited thread not owned by the instance.